### PR TITLE
Use local config files

### PR DESCRIPTION
**Pull request by alexander-demichev**
Closes https://github.com/fusor/cpma/issues/278

If ssh is not available, we ask user for source of config files, local or remote. If remote then we keep them same workflow we have right now, if local then just use survey to get paths listed here.
This PR also adds ability for user to choose how API client should be created, using cluster name from current context or prompt.
 
I will try to add tests for `survey` inputs later and update README after review.

Local files as source:
local
Remote files:
remote

### CODE REVIEW

File: pkg/env/env.go [hunk 1]
```diff
@@ -71,55 +65,117 @@ func InitConfig() error {
...
-			case err.Error() == "interrupt":
-				return errors.Wrap(err, "ctrl-C pressed. Exiting.")
-			default:
-				return errors.Wrap(err, "Error in creating config file")
-			}
+			return err
 		}
 
-		logrus.Debug("Can't read config file, all values were prompted and new config was asked to be created, err: ", readConfigErr)
+		viperConfig.Set("OutputDir", outPutDir)
```
Comment (arapov) [inline pkg/env/env.go]: a minor thing, worth to fix later: capital P in output looks really weird! :)

File: pkg/env/env.go [hunk 2]
```diff
@@ -264,6 +396,15 @@ func surveyCreateConfigFile() error {
 	return nil
 }
 
+func handleInterrupt(err error) error {
+	switch {
+	case err.Error() == "interrupt":
+		return errors.Wrap(err, "ctrl-C pressed. Exiting.")
```
Comment (arapov) [inline pkg/env/env.go]: All you want to write here is "Exiting", terminal will show ^C itself.

File: cmd/root.go [hunk 3]
```diff
@@ -55,8 +53,32 @@ func init() {
 	rootCmd.PersistentFlags().StringVarP(&env.Port, "ssh-port", "p", "", "OCP3 ssh port")
 
 	// Get config file from CLI argument an save to viper config
-	rootCmd.PersistentFlags().StringP("output-dir", "o", path.Dir("."), "set the directory to store extracted configuration.")
+	rootCmd.PersistentFlags().StringP("output-dir", "o", "", "set the directory to store extracted configuration.")
```
Comment (gildub) [inline cmd/root.go]: Why changing back to ""? Because "." fixes the issue to save current dir as default `OutputDir`

File: cmd/root.go [hunk 4]
```diff
@@ -55,8 +53,32 @@ func init() {
...
 	env.Config().BindPFlag("OutputDir", rootCmd.PersistentFlags().Lookup("output-dir"))
+
+	// Set log level from CLI argument
+	rootCmd.PersistentFlags().String("config-source", "", "source for OCP3 config files, accepted values: remote or local")
+	env.Config().BindPFlag("ConfigSource", rootCmd.PersistentFlags().Lookup("config-source"))
+
+	// Get crio config file location
+	rootCmd.PersistentFlags().String("crio-config", "", "path to crio config file")
```
Comment (gildub) [inline cmd/root.go]: Can we make the default OCP 3.x values the default value for CPMA?

Same for the other ones: master-config, node-config, etc.

Comment (alexander-demichev) [reply cmd/root.go]: Yes, default values are set only in case of usage ssh as source for config

Comment (alexander-demichev) [reply cmd/root.go]: https://github.com/fusor/cpma/pull/279/files#diff-1e94282afa1637126c37c2d26509a2d4R259

File: cmd/root.go (hunk 3 above)
Comment (alexander-demichev) [reply cmd/root.go]: +1

**Pull request merged**